submitit/core/utils.py

```diff
@@ -238,6 +240,56 @@ def cloudpickle_dump(obj: Any, filename: Union[str, Path]) -> None:
         cloudpickle.dump(obj, ofile, pickle.HIGHEST_PROTOCOL)
 
 
+class _MultiStreamWrapper:
+    """
+    One-to-many wrapper for IO write streams.
+    """
+
+    def __init__(self, streams: List[IO[str]]):
+        self._streams = streams
+
+    def write(self, content: str):
+        for stream in self._streams:
+            stream.write(content)
+
+    def flush(self):
+        for stream in self._streams:
+            stream.flush()
+
+
+def copy_streams(in_streams: List[IO[bytes]], out_streams: List[_MultiStreamWrapper]):
+    """
+    Using `select`, copy the content from the bytes `in_streams` streams,
+    to the matching out stream in `out_streams`.
+    """
+    assert len(in_streams) == len(out_streams)
+
+    # We must use the raw buffer, as otherwise this could mess up our calls to select.
```

Comment:
are you sure ? the concept of buffer is on python side, while select is a unix syscall. I believe that `file.fileno() == file.raw.fileno()`. Am I wrong ? Can they have different descriptors ?

Reply:
Yes but the issue is, if you do a read with a buffered file, maybe the buffered interface kept some of the data to itself (imagine there is a lot of data available, and we ask to read a smaller buffer than what is available), then it is possible data is available for  read (already buffered), but because `select` doesn't know about the buffered data, it will say no data is available, and if this was the very last chunk of data sent out, we will miss it, well at least only until the pipe gets closed.

Reply:
ok, I restored the conversion to raw buffer.